core/trino-main/src/main/java/io/trino/operator/aggregation/partial/SkipAggregationBuilder.java

```diff
@@ -0,0 +1,222 @@
+import io.trino.operator.WorkProcessor.ProcessState;
+import io.trino.operator.aggregation.AggregatorFactory;
+import io.trino.operator.aggregation.GroupedAggregator;
+import io.trino.operator.aggregation.builder.HashAggregationBuilder;
+import io.trino.spi.Page;
+import io.trino.spi.block.Block;
+import io.trino.spi.block.BlockBuilder;
+import io.trino.spi.block.LongArrayBlock;
+
+import java.util.List;
+import java.util.Optional;
+
+import static com.google.common.base.Preconditions.checkArgument;
+import static com.google.common.collect.ImmutableList.toImmutableList;
+import static java.util.Objects.requireNonNull;
+
+/**
+ * {@link HashAggregationBuilder} that does not aggregate input rows at all.
+ * It passes the input pages, augmented with initial accumulator state to the output.
+ * It can only be used at the partial aggregation step as it relies on rows be aggregated at the final step.
+ */
+public class SkipAggregationBuilder
+        implements HashAggregationBuilder
+{
+    private final LocalMemoryContext memoryContext;
+    private final List<GroupedAggregator> groupedAggregators;
+    private volatile Page currentPage;
+    private final int[] hashChannels;
+
+    public SkipAggregationBuilder(
+            List<Integer> groupByChannels,
+            Optional<Integer> inputHashChannel,
+            List<AggregatorFactory> aggregatorFactories,
+            LocalMemoryContext memoryContext)
+    {
+        this.memoryContext = requireNonNull(memoryContext, "memoryContext is null");
+        this.groupedAggregators = requireNonNull(aggregatorFactories, "aggregatorFactories is null")
+                .stream()
+                .map(AggregatorFactory::createGroupedAggregator)
+                .collect(toImmutableList());
+        this.hashChannels = new int[groupByChannels.size() + (inputHashChannel.isPresent() ? 1 : 0)];
+        for (int i = 0; i < groupByChannels.size(); i++) {
+            hashChannels[i] = groupByChannels.get(i);
+        }
+        inputHashChannel.ifPresent(inputHash -> hashChannels[groupByChannels.size()] = inputHash);
+    }
+
+    @Override
+    public Work<?> processPage(Page page)
+    {
+        checkArgument(currentPage == null);
+        return new Work<Void>()
+        {
+            @Override
+            public boolean process()
+            {
+                checkArgument(currentPage == null);
+                currentPage = page;
+                return true;
+            }
+
+            @Override
+            public Void getResult()
+            {
+                return null;
+            }
+        };
+    }
+
+    @Override
+    public WorkProcessor<Page> buildResult()
+    {
+        WorkProcessor<Page> result = WorkProcessor.create(new SinglePageResult(currentPage));
+        currentPage = null;
+        return result;
+    }
+
+    @Override
+    public boolean isFull()
+    {
+        return currentPage != null;
+    }
+
+    @Override
+    public void updateMemory()
```

Comment:
nit: in reality we could actually skip updating memory usage because it's just a single page that is consumed immediately, but let's keep it since it doesn't seem to cause any regressions